### PR TITLE
Added clarification for match expression multiline case

### PR DESCRIPTION
**Pull request by Lanayx**
Discussed and approved in https://github.com/fsharp/fslang-design/issues/650

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/fsharp/style-guide/formatting.md](https://github.com/dotnet/docs/blob/f0bf252ec90be87e767f06608fe0ee3ed8ef2569/docs/fsharp/style-guide/formatting.md) | [F# code formatting guidelines](https://review.learn.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting?branch=pr-en-us-40762) |

### CODE REVIEW

File: docs/fsharp/style-guide/formatting.md [hunk 1]
```diff
@@ -1092,16 +1092,24 @@ match l with
     | [] -> failwith "Couldn't find David"
 ```
 
-If the expression on the right of the pattern matching arrow is too large, move it to the following line, indented one step from the `match`/`|`.
+If one of the expressions on the right of the pattern matching arrow is too large, move it to the following line, indented one step from the `match`/`|`. Move all the other expressions to the next line as well for consistency.
```
Comment (T-Gro) [inline docs/fsharp/style-guide/formatting.md:1095]: I don't interpret Don's last comment in the suggestion as setting this as the default, but rather an option.
I think the wording should reflect that (or the suggestion discussion continued about the default-ness of this)

**Comment by alandryz83** [comment]
Sent from my iPhoneOn May 6, 2024, at 4:07 AM, Tomas Grosup ***@***.***> wrote:﻿
@T-Gro commented on this pull request.



In docs/fsharp/style-guide/formatting.md:
> @@ -1092,16 +1092,24 @@ match l with
     | [] -> failwith "Couldn't find David"
 ```
 
-If the expression on the right of the pattern matching arrow is too large, move it to the following line, indented one step from the `match`/`|`.
+If one of the expressions on the right of the pattern matching arrow is too large, move it to the following line, indented one step from the `match`/`|`. Move all the other expressions to the next line as well for consistency.

I don't interpret Don's last comment in the suggestion as setting this as the default, but rather an option.
I think the wording should reflect that (or the suggestion discussion continued about the default-ness of this)

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

**Comment by alandryz83** [comment]
Sent from my iPhoneOn May 6, 2024, at 4:07 AM, Amanda Landry ***@***.***> wrote:﻿Sent from my iPhoneOn May 6, 2024, at 4:07 AM, Tomas Grosup ***@***.***> wrote:﻿
@T-Gro commented on this pull request.



In docs/fsharp/style-guide/formatting.md:
> @@ -1092,16 +1092,24 @@ match l with
     | [] -> failwith "Couldn't find David"
 ```
 
-If the expression on the right of the pattern matching arrow is too large, move it to the following line, indented one step from the `match`/`|`.
+If one of the expressions on the right of the pattern matching arrow is too large, move it to the following line, indented one step from the `match`/`|`. Move all the other expressions to the next line as well for consistency.

I don't interpret Don's last comment in the suggestion as setting this as the default, but rather an option.
I think the wording should reflect that (or the suggestion discussion continued about the default-ness of this)

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

**Comment by alandryz83** [comment]
Sent from my iPhoneOn May 5, 2024, at 3:07 PM, Vladimir Shchur ***@***.***> wrote:﻿Discussed and approved in fsharp/fslang-design#650



You can view, comment on, or merge this pull request online at:
  https://github.com/dotnet/docs/pull/40762

Commit Summary

  f0bf252  Added clarification for match expression multiline case


File Changes  (1 file)

  
    M
    docs/fsharp/style-guide/formatting.md
    (12)
  


Patch Links:

  https://github.com/dotnet/docs/pull/40762.patch
  https://github.com/dotnet/docs/pull/40762.diff

File: docs/fsharp/style-guide/formatting.md (hunk 1 above)
Comment (Lanayx) [reply docs/fsharp/style-guide/formatting.md:1095]: @dsyme set the label _style-guide-decided_, so I regard it as a marker that style guide is decided (final decision). As I understand, Don was not sure about tooling changes needed to support this style, but it should not affect the style and the recommendation itself.

Comment (T-Gro) [reply docs/fsharp/style-guide/formatting.md:1095]: The label is fine.
What I do not see confirmed is that "defaultness" of this, I interpret the context of the suggestion that this should still remain a possible option only.

Comment (Lanayx) [reply docs/fsharp/style-guide/formatting.md:1095]: Ok, looks like it can't be resolved without Don

Comment (abonie) [reply docs/fsharp/style-guide/formatting.md:1095]: I agree with @T-Gro , here is what don said
> I agree with @isaacabraham that at option (4) listed above should be **an option**.

(emphasis mine)
cc: @vzarytovskii

**Comment by Lanayx** [comment]
Closing PR as it appeared that assumption of approved style was wrong

**Pull request closed without merging**